### PR TITLE
feat: Add article about turborepo

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

File: blog/2022-02-08-turborepo-overhyped.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Setting up
+
+I was able to understand Turborepo docs quite quickly (as its really short so far :D). It took me a bit of trial and error to set everything up correctly as I needed to wrap my head around how exactly the caching works. I recommend spending some time experimenting with changes in different projects and checking if it works correctly as it takes some time to wrap around your brain. However at the end, it is very satisfying, when you see colorful label "FULL TURBO", when everything is taken from cache and no real build is necessary. 
+
+One little caveat was that we are using lerna, which has it's own definition for workspaces, however Turborepo requires the definition to be in `package.json`, which causes issues with `npm install` command as npm now also supports workspaces, but is installing dependencies differently than we need. I didn't find a simple way how to get around this issue so at the end I use custom command for module installation, which first removes workspaces from root package.json, then performs all the installations with lerna and then returns the package.json to original state, not ideal but managable.
```
Comment (marketachalupnikova) [inline blog/2022-02-08-turborepo-overhyped.md:30]: Lerna

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:30]: its own

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:30]: however,

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:30]: manageable

File: blog/2022-02-08-turborepo-overhyped.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Where the real issues started
+
+During my experiments I've noticed that sometimes not all dependant projects are rebuild as they should be. I found already reported issue about it and amazingly the next day it was fixed. Wow. I wanned to upgrade to next version but wait ... that's weird some error appeared during an installation.
```
Comment (marketachalupnikova) [inline blog/2022-02-08-turborepo-overhyped.md:34]: experiments,

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:34]: dependent

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:34]: rebuilt

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:34]: an already

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:34]: I wanted....the next version

File: blog/2022-02-08-turborepo-overhyped.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+Hmm, I found another reported error - it seems like it's broken when you install it with npm (yarn and pnpm works), thats unfortunate 😔. Luckily it was fixed in about 4 hours, so now everything worked fine and I've prepared and tested everything and submited it for a review. However my collegue writes me back in two hours that the process it hitting his CPU running on 100% (one core) and his machine is constantly overheating even when he's in development mode and not making any changes to the code! Another issue (already existing on github), Agh!!
```
Comment (marketachalupnikova) [inline blog/2022-02-08-turborepo-overhyped.md:36]: that's unfortunate

File: blog/2022-02-08-turborepo-overhyped.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+<!--truncate-->
+
+## What exactly is Turborepo and why we need it?
+
+Turborepo is a build system for monorepos (multiple projects in one repository). We use monorepo in Tolgee, because we have multiple SDKs for different frameworks as separate NPM packages and it is very convenient to test and release them together, so monorepo is an ideal fit.
+
+As with every monorepo we struggle to keep our building and testing pipeline fast. Turborepo is designed for this. Simply put, it keeps a cache of previous builds for each individual project (workspace) and if nothing changed in that project, it will just use the cache instead of rebuilding. It also automatically detects dependencies between the projects so it knows that if dependency of one project changed that project needs to be rebuild too.
+
+It can be also used for testing as you can only run tests in projects which are affected by changes and it treats standard output as part of a cache so even if some test don't run Turborepo will output cached logs, so you see all the test results every time.
```
Comment (JanCizmar) [inline blog/2022-02-08-turborepo-overhyped.md:22]: How does it detect which changes are related to the particular test?

File: blog/2022-02-08-turborepo-overhyped.md (hunk 3 above)
Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:36]: submitted

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:36]: However,

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:36]: colleague

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:36]: is hitting

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:36]: GitHub

File: blog/2022-02-08-turborepo-overhyped.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+## Where the real issues started
+
+During my experiments I've noticed that sometimes not all dependant projects are rebuild as they should be. I found already reported issue about it and amazingly the next day it was fixed. Wow. I wanned to upgrade to next version but wait ... that's weird some error appeared during an installation.
+
+Hmm, I found another reported error - it seems like it's broken when you install it with npm (yarn and pnpm works), thats unfortunate 😔. Luckily it was fixed in about 4 hours, so now everything worked fine and I've prepared and tested everything and submited it for a review. However my collegue writes me back in two hours that the process it hitting his CPU running on 100% (one core) and his machine is constantly overheating even when he's in development mode and not making any changes to the code! Another issue (already existing on github), Agh!!
+
+Even when the issue was fixed the next day, I wasn't really that confident about Turborepo as even though it was in version 1.0.xx it seemed more like beta or alpha version. However now everything worked so PR was merged.
```
Comment (marketachalupnikova) [inline blog/2022-02-08-turborepo-overhyped.md:38]: However,

File: blog/2022-02-08-turborepo-overhyped.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+A few hours later I've noticed a weird thing. Our our available pipelines were taken and I was not able to run a new one. It turned out that when some build failed Turborepo didn't finish the process but it was stucked and ran forewer. I had to manually stop all pipelines and rollback the release as this was a major problem.
```
Comment (marketachalupnikova) [inline blog/2022-02-08-turborepo-overhyped.md:40]: Our (2 times)

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:40]: stuck....forever

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:40]: roll back

File: blog/2022-02-08-turborepo-overhyped.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+## Maybe I made a mistake
+
+Hmm, seems like Turborepo is not in production ready state. I was really pissed as I've already spent about 4 days with this and now I'm back where I was at the beginning. I've tried to switch to Nx, but compared to the Turborepo setup everything was much more complicated - it has all the functions we need, but it's really blown and not elegant at all.
```
Comment (marketachalupnikova) [inline blog/2022-02-08-turborepo-overhyped.md:44]: production-ready

File: blog/2022-02-08-turborepo-overhyped.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+### Or not?
+
+I've let everything as it was and let the PR open for one month. Now I've returned to it and Turborepo is in version 1.1.xx and everything seems to work as expected. Ok, our pipeline is much faster and we now have much more elegant way how to run dev mode locally. However I still better run our main release pipeline without any caching as I want to be sure that something is not getting cached when it shouldn't be as that really could be a big problem.
+
+## My conclusion
+
+Turborepo is a really cool project. And it's not just cool, it's really necessary - there clearly was missing some tool like this as monorepos are more and more popular. Everyone was tweeting about using Turborepo is great and that it made his pipeline 10x faster, but nobody was mentioning these issues. I think that the hype around it caused me to use it prematurely and lose a valuable time and at the and be a bitter about it.
```
Comment (marketachalupnikova) [inline blog/2022-02-08-turborepo-overhyped.md:52]: lose valuable time

Comment (marketachalupnikova) [reply blog/2022-02-08-turborepo-overhyped.md:52]: in the end be bitter

File: blog/2022-02-08-turborepo-overhyped.md (hunk 6 above)
Comment (JanCizmar) [inline blog/2022-02-08-turborepo-overhyped.md:40]: Our our

Comment (JanCizmar) [reply blog/2022-02-08-turborepo-overhyped.md:40]: our available paralel pipelines -> our available parallel workflow runs on GitHub actions?

**Review by JanCizmar — CHANGES_REQUESTED** [review]

File: blog/2022-02-08-turborepo-overhyped.md (hunk 4 above)
Comment (stepan662) [reply blog/2022-02-08-turborepo-overhyped.md:22]: When there are changes in the project or in the project that is dependency

**Pull request merged**